src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/config/PluginSettings.java

```diff
@@ -271,4 +295,21 @@ private void loadMetricsDBFilesCleanupEnabled() {
       shouldCleanupMetricsDBFiles = true;
     }
   }
+
+  private void loadBatchMetricsRetentionPeriodMinutesFromConfig() {
+    if (!settings.containsKey(BATCH_METRICS_RETENTION_PERIOD_MINUTES)) {
+      return;
+    }
+
+    try {
+      long parsedRetentionPeriod = Long.parseLong(settings.getProperty(BATCH_METRICS_RETENTION_PERIOD_MINUTES));
+      if (parsedRetentionPeriod <= 0 || parsedRetentionPeriod > BATCH_METRICS_RETENTION_PERIOD_MINUTES_LIMIT) {
+        throw new InvalidParameterException();
```

Comment:
That is what this code does. batchMetricsRetentionPeriodMinutes was assigned a default value in the ctor. The code here only updates that value if it was able to properly parse a new valid value from the properties file.

Reply:
Right, so you don't need to throw a InvalidParameterException, you can simple return ?

Reply:
I am throwing an InvalidParameterException so I can catch it (along with the NumberFormatException that may be thrown by Long.parseLong) and log the error.

Reply:
But yeah, it seems like it might be more appropriate to just give more user-friendly information about the value being out of range, like loadMetricsDeletionIntervalFromConfig is.